packages/microlinkjs-wp/includes/class-microlinkjs.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/**
+ * The file that defines the core plugin class
+ *
+ * A class definition that includes attributes and functions used across both the
+ * public-facing side of the site and the admin area.
+ *
+ * @link       https://microlink.io/
+ * @since      1.0.0
+ *
+ * @package    Microlinkjs
+ * @subpackage Microlinkjs/includes
+ */
+
+/**
+ * The core plugin class.
+ *
+ * This is used to define internationalization, admin-specific hooks, and
+ * public-facing site hooks.
+ *
+ * Also maintains the unique identifier of this plugin as well as the current
+ * version of the plugin.
+ *
+ * @since      1.0.0
+ * @package    Microlinkjs
+ * @subpackage Microlinkjs/includes
+ * @author     Microlink <hello@microlink.io>
+ */
+class Microlinkjs {
+
+	/**
+	 * The loader that's responsible for maintaining and registering all hooks that power
+	 * the plugin.
+	 *
+	 * @since    1.0.0
+	 * @access   protected
+	 * @var      Microlinkjs_Loader    $loader    Maintains and registers all hooks for the plugin.
+	 */
+	protected $loader;
+
+	/**
+	 * The unique identifier of this plugin.
+	 *
+	 * @since    1.0.0
+	 * @access   protected
+	 * @var      string    $plugin_name    The string used to uniquely identify this plugin.
+	 */
+	protected $plugin_name;
+
+	/**
+	 * The current version of the plugin.
+	 *
+	 * @since    1.0.0
+	 * @access   protected
+	 * @var      string    $version    The current version of the plugin.
+	 */
+	protected $version;
+
+	/**
+	 * Define the core functionality of the plugin.
+	 *
+	 * Set the plugin name and the plugin version that can be used throughout the plugin.
+	 * Load the dependencies, define the locale, and set the hooks for the admin area and
+	 * the public-facing side of the site.
+	 *
+	 * @since    1.0.0
+	 */
+	public function __construct() {
+		if ( defined( 'PLUGIN_NAME_VERSION' ) ) {
+			$this->version = PLUGIN_NAME_VERSION;
+		} else {
+			$this->version = '1.0.0';
```

Comment:
humm, can we read this value from a `package.json`? Because when lerna create a new tag it increments the version of the file.

Reply:
I _think_ we might be able to actually, loading the `package.json` within the php and just reading the version node from it.

I'll see what I can do 👍